packages/eslint-plugin/tests/rules/use-unknown-in-catch-callback-variable.test.ts

```diff
@@ -104,6 +112,57 @@ ruleTester.run('use-unknown-in-catch-callback-variable', rule, {
         console.log(find);
       });
     `,
+    'Promise.resolve.then();',
+    'Promise.resolve().then(() => {});',
+    `
+      declare const singleTupleArg: [() => void];
+      Promise.resolve().then(...singleTupleArg, (error: unknown) => {});
+    `,
+    `
+      declare const arrayArg: (() => void)[];
+      Promise.resolve().then(...arrayArg, error => {});
+    `,
+    `
+declare let iPromiseImAPromise: Promise<any>;
+declare const catchArgs: [(x: any) => void];
+iPromiseImAPromise.catch(...catchArgs);
+    `,
+    `
+declare const catchArgs: [
+  string | (() => never) | ((x: string) => void),
+  number,
+];
+Promise.reject(new Error()).catch(...catchArgs);
+    `,
+    `
+declare const you: [];
+declare const cannot: [];
+declare const fool: [];
+declare const me: [(x: Error) => void] | undefined;
+Promise.resolve(undefined).catch(...you, ...cannot, ...fool, ...me!);
+    `,
+    `
+declare const really: undefined[];
+declare const dumb: [];
+declare const code: (x: Error) => void;
+Promise.resolve(undefined).catch(...really, ...dumb, code);
+    `,
+    `
+declare const x: ((x: any) => string)[];
+Promise.resolve('string promise').catch(...x);
+    `,
+    `
+declare const x: any;
+Promise.resolve().catch(...x);
+    `,
+    `
+declare const thenArgs: [() => {}, (err: any) => {}];
+Promise.resolve().then(...thenArgs);
+    `,
+    `
+      let method = 'catch';
+      Promise.resolve()[method]((error: Error) => {});
```

Comment:
This test covers the case when `staticMemberAccessKey` is `null`, getting us to 100% test coverage 👍 

However, while writing this test, I discovered that this rule, both in this branch and on `main`, does not report on
```
const x = 'catch';
Promise.resolve()[x]((err: any) => { });
```
even though some of the code comments in this file make it sound like it should.
Since that is unrelated to this change, I'm going to open that up as a separate issue and PR.